spine/runSpineClients.js

```diff
@@ -1,105 +1,127 @@
-'use strict';
-var Getopt = require('node-getopt');
-var efc = require ('./simpleNativeConnection');
+const fs = require('fs');
+const Getopt = require('node-getopt'); // eslint-disable-line
+const Spine = require('./Spine.js');
 
-var getopt = new Getopt([
-  ['s' , 'stream-config=ARG'             , 'file containing the stream config JSON'],
-  ['t' , 'time=ARG'                      , 'interval time to show stats (default 10 seconds)'],
-  ['h' , 'help'                          , 'display this help']
+const getopt = new Getopt([
+  ['s', 'stream-config=ARG', 'file containing the stream config JSON (default is spineClientsConfig.json)'],
+  ['i', 'interval=ARG', 'interval time to show stats (default 10 seconds)'],
+  ['t', 'total-intervals=ARG', 'total test intervals (default 10 intervals)'],
+  ['o', 'output=ARG', 'output file for the stat digest (default is testResult.json)'],
+  ['h', 'help', 'display this help'],
 ]);
+const opt = getopt.parse(process.argv.slice(2));
 
-var opt = getopt.parse(process.argv.slice(2));
+let streamConfig = 'spineClientsConfig.json';
+let statsIntervalTime = 10000;
+let totalStatsIntervals = 10;
+let statOutputFile = 'testResult.json';
+const optionKeys = Object.keys(opt.options);
 
-var streamConfig;
-
-var statsInterval = 10000;
-
-for (var prop in opt.options) {
-    if (opt.options.hasOwnProperty(prop)) {
-        var value = opt.options[prop];
-        switch (prop) {
-            case 'help':
-                getopt.showHelp();
-                process.exit(0);
-                break;
-            case 'stream-config':
-                streamConfig = value;
-                break;
-            case 'time':
-                statsInterval = value * 1000;
-                break;
-            default:
-                console.log('Default');
-                break;
-        }
-    }
-}
-
-if (!streamConfig){
-    streamConfig = 'spineClientsConfig.json';
-}
+optionKeys.forEach((key) => {
+  const value = opt.options[key];
+  switch (key) {
+    case 'help':
+      getopt.showHelp();
+      process.exit(0);
+      break;
+    case 'stream-config':
+      streamConfig = value;
+      break;
+    case 'interval':
+      statsIntervalTime = value * 1000;
+      break;
+    case 'totalIntervals':
+      totalStatsIntervals = value;
+      break;
+    case 'output':
+      statOutputFile = value;
+      break;
+    default:
+      console.log('Default');
+      break;
+  }
+});
 
 console.log('Loading stream config file', streamConfig);
-streamConfig = require('./' + streamConfig);
-
-if (streamConfig.publishConfig){
-    var streamPublishConfig = {
-        publishConfig: streamConfig.publishConfig,
-        serverUrl: streamConfig.basicExampleUrl
-    };
+streamConfig = require(`./${streamConfig}`); // eslint-disable-line
```

Comment:
there are other ways to read the config file better than require, we can fix it in the future.